crates/net/network-api/src/lib.rs

```diff
@@ -9,13 +9,20 @@
 //!
 //! Provides abstractions for the reth-network crate.
 
-use reth_primitives::NodeRecord;
+use async_trait::async_trait;
+use reth_primitives::{NodeRecord, H256, U256};
+use serde::{Deserialize, Serialize};
 use std::net::SocketAddr;
+use tokio::sync::oneshot;
 
 /// Provides general purpose information about the network.
+#[async_trait]
 pub trait NetworkInfo: Send + Sync {
     /// Returns the [`SocketAddr`] that listens for incoming connections.
     fn local_addr(&self) -> SocketAddr;
+
+    /// Returns the current status of the network being ran by the local node.
+    async fn network_status(&self) -> Result<NetworkStatus, oneshot::error::RecvError>;
```

Comment:
To help make the error more generic, we could add an associated error type here, something like:
```rust
#[async_trait]
pub trait NetworkInfo: Send + Sync {
    /// Associated error type for the network implementation
    type Error: Send + Sync + Error;

    /// Returns the [`SocketAddr`] that listens for incoming connections.
    fn local_addr(&self) -> SocketAddr;

    /// Returns the current status of the network being ran by the local node.
    async fn network_status(&self) -> Result<NetworkStatus, Self::Error>;
```

This way we don't have to specify implementation-specific error types in the `NetworkInfo` trait, even if there are other error types that the `NetworkHandle` returns

Reply:
Ahhh that looks neat! done in https://github.com/paradigmxyz/reth/pull/923/commits/36ff7736474f649a43bd9d9e81881b91b9e4a3f5